src/Costura.Template/ILTemplateWithTempAssembly.cs

```diff
@@ -26,6 +27,20 @@ public static void Attach()
             return;
         }
 
+        var currentDomain = AppDomain.CurrentDomain;
+
+        // Make sure the target framework is set in order not to interfere with AppContext switches initialization
+        // See https://github.com/Fody/Costura/issues/633 for full explanation
+        if (AppContext.TargetFrameworkName == null)
```

Comment:
Just [checked](https://docs.microsoft.com/en-us/dotnet/api/system.appcontext?view=net-5.0), this was introduced in .NET 4.6. This means the minimum required version will move to this when using this template:

![image](https://user-images.githubusercontent.com/1246444/104888593-3cb90d80-596d-11eb-963e-433cca65aac8.png)

Reply:
Actually it will move to [.NET Framework 4.7](https://docs.microsoft.com/en-us/dotnet/api/system.appcontext.targetframeworkname). The `AppContext` class was introduced in 4.6 but the `TargetFrameworkName` was added later in 4.7.

I first tried to use `AppDomain.CurrentDomain.SetupInformation.TargetFrameworkName` but it was not available since the `Costura.Template` project targets .NET Standard 2.0. This is why I used `AppContext.TargetFrameworkName` instead.

Maybe I should use reflection instead? I haven't tested what happens if this code runs on .NET Framework 4.5. I guess it will throw in the module initializer. 😕